core/tests/conftest.py

```diff
@@ -15,6 +15,15 @@ def reset_global_state():
     toga_window._window_count = -1
 
 
+@pytest.fixture(autouse=True)
+def no_dangling_tasks():
+    """Ensure any tasks for the test were removed when the test finished."""
+    yield
+    if toga.App.app:
+        tasks = toga.App.app._running_tasks
+        assert not tasks, f"the app has dangling tasks: {tasks}"
```

Comment:
I tried to implement this as an additional task callback....but apparently Exceptions raised in a task callback in pytest just disappear...

Reply:
Yeah - this is the bug I saw side-on when testing the document app/status icon stuff. There were exceptions being raised, but hidden. Not sure if this is a configuration issue with Pytest(-async) though...